dcos/client.go

```diff
@@ -38,9 +53,100 @@ func NewClientWithOptions(httpClient *http.Client, config *Config) (*Client, err
 		return nil, fmt.Errorf("config cannot be nil")
 	}
 
+	iamClient, err := newIamClient(config.URL(), httpClient)
+	if err != nil {
+		return nil, fmt.Errorf("could not create IAM client: %s", err)
+	}
+
+	secretsClient, err := newSecretsClient(config.URL(), httpClient)
+	if err != nil {
+		return nil, fmt.Errorf("could not create Secrets client: %s", err)
+	}
+
+	cosmosClient, err := newCosmosClient(config.URL(), httpClient)
+	if err != nil {
+		return nil, fmt.Errorf("could not create Cosmos client: %s", err)
+	}
+
 	return &Client{
 		HTTPClient: httpClient,
 		Config:     config,
-		UserAgent:  fmt.Sprintf("%s(%s)", ClientName, Version),
+		IAM:        iamClient,
+		Secrets:    secretsClient,
+		Cosmos:     cosmosClient,
 	}, nil
 }
+
+func (c *Client) Login(username, password string) (string, error) {
```

Comment:
Given that we have different login methods (username/password, service account, oidc/saml), in the future either we say uid/password is the default and keep it named `Login`, then we'll add eg. `ServiceLogin`, we could also make the arguments more generic instead.